xdsl/transforms/individual_rewrite.py

```diff
@@ -94,30 +77,34 @@ class ApplyIndividualRewritePass(ModulePass):
 
     name = "apply-individual-rewrite"
 
-    matched_operation_index: int | None = None
-    operation_name: str | None = None
-    pattern_name: str | None = None
+    matched_operation_index: int = field()
+    operation_name: str = field()
+    pattern_name: str = field()
```

Comment:
I had been thinking that it might be more ergonomic here to just store the pattern directly here, instead of storing the name of the pattern and then looking it up, as we currently duplicate the lookup. Thoughts?

Reply:
fair enough, that makes a lot of sense